### DISCUSSION TITLE
Ignore rules in git data source

### DISCUSSION THREAD

**Discussion by chuegel** [netbox-community/netbox#13611] [Help Wanted!]
Good day,

since the ignore rules do not follow typical gitignore pattern, I have a hard time to exclude everything from a git repo except one folder.
Things like `'[!dirname]'`  doesn't work
Does anyone have the correct syntax to achieve that?
Any help is much appreciated.

**Comment by chuegel** [comment]
Is there a reason why not use `glob` instead of `fnmatch`?

**Reply by ziggekatten** [reply]
Also struggling with this. Can't  exclude directories

**Reply by kkthxbye-code** [reply]
Feel free to create a feature request.

**Comment by eliezerlp** [comment]
In the [Netbox code](https://github.com/netbox-community/netbox/blob/5b2f29480a42e9c815ca1f50403c115cc89b6538/netbox/core/models/data.py#L221) you can see that the `fnmatch` patterns are only applied to the filenames, not the path+filename.
